pyinstrument/renderers/console.py

```diff
@@ -137,11 +124,65 @@ def render_frame(self, frame: Frame, indent: str = "", child_indent: str = "") -
 
         return result
 
-    def frame_proportion_of_total_time(self, frame: Frame):
-        return frame.time / self.root_frame.time
+    def render_frame_flat(self, frame: Frame) -> str:
+        def walk(frame: Frame):
+            frame_id_to_time[frame.identifier] = (
+                frame_id_to_time.get(frame.identifier, 0) + frame.total_self_time
+            )
+
+            frame_id_to_frame[frame.identifier] = frame
+
+            for child in frame.children:
+                walk(child)
+
+        frame_id_to_time: Dict[str, float] = {}
+        frame_id_to_frame: Dict[str, Frame] = {}
+
+        walk(frame)
+
+        id_time_pairs: List[Tuple[str, float]] = sorted(
+            frame_id_to_time.items(), key=(lambda item: item[1]), reverse=True
+        )
+
+        # remove nodes that represent less than 0.1% of the total time
+        id_time_pairs = [pair for pair in id_time_pairs if pair[1] / self.root_frame.time > 0.001]
```

Comment:
Should this be affected by `--show-all`?

Reply:
yeah good catch! i'll chuck that in too